### ISSUE TITLE
Self hosted interface

### ISSUE THREAD

**Issue by lathanbritz** [EvernodeXRPL/evernode-sdk#90]
Now that the nodes require a SSL is there a how to showing a HTML page that can be attached on a contract to act as a self hosted interface for the contract.....

**Comment by ravinsp** [comment]
@lathanbritz Neat use-case. I remember way back @RichardAH also had a vision to have mini web pages exposed through Evernode contracts but the web page was bootstrapped via the websocket connection first. Your idea is about bootstrapping the websocket connections via a web page so the first request must happen via HTTP protocol.

The problem is the contract only exposes websocket port to the outside world, meaning a standard web browser HTTP request won't be compatible with the websocket interface. The contract could program in a websocket message which sends a HTML content response. But in order to download the HTML content and show it on the browser, you would still need a websocket client running first. This would defeat the purpose you're trying to do this in the first place.

In short, your idea is currently NOT possible. However sashimono could be made to add support for this in future by means of the contract instance providing this HTML page beforehand to sashimono, and sashimono serving all such web pages of the contracts on port 443 on the host via some HTTP url pattern. (eg: `https://www.myhost.com/<contract_instance_identifier>)

**Comment by lathanbritz** [comment]
I would suggest exploring IPFS protocol here if it get evaluated later. Evernode it's self could become a IPFS type solution, where by a contract gets a hash much like a file (should change if contract changes/updates). This way the existing IPFS support in browsers etc could be leveraged.

Anyhow an idea / option down the road.
